numpy/distutils/exec_command.py

```diff
@@ -249,14 +249,28 @@ def _exec_command(command, use_shell=None, use_tee = None, **env):
         proc = subprocess.Popen(command, shell=use_shell, env=env,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.STDOUT,
-                                universal_newlines=True)
+                                universal_newlines=False)
+        # universal_newlines is set to False so that communicate()
+        # will return bytes. We need to decode the output ourselves
+        # so that Python will not raise a UnicodeDecodeError when
+        # it encounters an invalid character; rather, we simply replace it
```

Comment:
Comments usually go before the think they describe.

FWIW, we could get around all this by using `errors='replace'` in Popen in python3.6